### DISCUSSION TITLE
Unexpected type error with assign

### DISCUSSION THREAD

**Discussion by bstro** [statelyai/xstate#1926] [Q&A]
Hi there,
I am fairly new to XState and working my way through examples. I'm trying to create my own simple state machine to represent a Selection service, which can be "active" with a selected object (stored in a context), or "inactive". Here's a [code sandbox](https://codesandbox.io/s/silly-rosalind-b8jj8?file=/src/index.ts).

I am trying to configure an initial `inactive` state that can receive a `SELECT` event with an object payload.  However, I see a type error and try as I might, I can't seem to get it to clear and I'm a bit stuck. 🙇 

![image](https://user-images.githubusercontent.com/627290/107550612-19593b00-6b86-11eb-8aa1-f4bb8f16689f.png)

![image](https://user-images.githubusercontent.com/627290/107550943-84a30d00-6b86-11eb-971f-3916146dd325.png)

**Comment by callmiy** [comment]
```typescript
{
  [selectionStates.INACTIVE]: {
      on: {
        [selectionEventTypes.SELECT as string]: {
          target: selectionStates.ACTIVE,
          actions: assign<
            Context,
            ReturnType<typeof selectionActions["SELECT"]>
          >({
            card: (_ctx, event) => some(event.card)
          })
        }
      }
  }
}
```

The fix is type assertion here: `[selectionEventTypes.SELECT as string]`
